File: src/two-sum/two_sum2.cpp
/*
 * @file two_sum.cpp
 * @author Akagi201
 * @date 2015/01/28
 *
 * https://oj.leetcode.com/problems/two-sum-ii-input-array-is-sorted/
 */

#include <vector>

using namespace std;

vector<int> twoSum(vector<int> &numbers, int target) {
  vector<int> result;
  int low = 0;
  int high = numbers.size() - 1;
  while (low < high) {
    if (numbers[low] + numbers[high] == target) {
      result.push_back(low + 1);
      result.push_back(high + 1);
      return result;
    } else {
      numbers[low] + numbers[high] > target ? high-- : low++;
    }
  }

  return result;
}

File: python/two-sum/two_sum2.py
# @file two_sum2.py
# @author Akagi201
# @date 2015/07/28
# https://oj.leetcode.com/problems/two-sum-ii-input-array-is-sorted/

# Time: O(n)
# Space: O(1)


class Solution:
    # @param {integer[]} numbers
    # @param {integer} target
    # @return {integer[]}

    def twoSum(self, numbers, target):
        start, end = 0, len(numbers) - 1

        while start != end:
            sum = numbers[start] + numbers[end]
            if sum > target:
                end -= 1
            elif sum < target:
                start += 1
            else:
                return [start + 1, end + 1]
    tests = [
        ([2, 7, 11, 15], 9, [1, 2])
    ]
